### PR TITLE
CORE-14484 Implement interop group management

### PR DESCRIPTION
**Pull request by ac101m**
# Changes
- Introduce `holdingIdentityShortHash` into interop identity avro schema

### CODE REVIEW

File: data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc [hunk 1]
```diff
@@ -15,6 +15,11 @@
       "doc": "Interop identity x500 name.",
       "name": "x500Name",
       "type": "string"
+    },
+    {
+      "doc": "Short hash of holding identity to which this interop identity corresponds.",
+      "name": "holdingIdentityShortHash",
```
Comment (szymonsztuka) [inline data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: If this is a value to distinguish whenever it is a) my alias versus b) an alias to someone I see, than this should be just a boolean flag. For aliases which are not "mine" ( b) ) the field is null, which is (n-1)*100/n % of times, where n is number of entries in the interop group.

Comment (ac101m) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: I strongly disagree. If this object is received outside the context of a holding identity or virtual node, then an "is my holding identity" flag loses all meaning. If we need to get "our" interop identity within a group, then the bookkeeping for that should be added to the new class I introduced into the cache.

Comment (szymonsztuka) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: but the record on Kafka topic is always under the key, this Avro type is only "Transport Object", when you read off the topic then you can create more proper business domain object.

Comment (ac101m) [reply data/avro-schema/src/main/resources/avro/net/corda/data/interop/InteropIdentity.avsc]: Oh I see, just for the avro type. I was imagining this being applied throughout, yeah I suppose that makes sense. Will update this.

**Review by yarivus — APPROVED** [review]
LGTM

**Review by jmacmahonr3 — APPROVED** [review]
LGTM

**Pull request merged**